lambda/lambda.mli

```diff
@@ -127,15 +127,11 @@ type primitive =
   | Psetfield_computed of immediate_or_pointer * initialization_or_assignment
   | Pfloatfield of int * field_read_semantics * locality_mode
   | Pufloatfield of int * field_read_semantics
-  | Pmixedfield of
-      int * mixed_block_read * mixed_block_shape * field_read_semantics
-  (* [Pmixedfield] is an access to either the flat suffix or value prefix of a
-     mixed record.
-  *)
+  | Pmixedfield of int * mixed_block_shape_with_locality_mode
+      * field_read_semantics
```

Comment:
I think a comment here and on `Psetmixedfield` clarifying whether this `int` is in terms of the source or the reordered block would be helpful.